src/o.js

```diff
@@ -48,85 +48,89 @@ o.prototype.setAttributes = function (attributes) {
     return this.attr(attributes);
 }
 
-o.prototype.attr = function(attrs) {
+o.prototype.attr = function (attrs) {
     if (Array.isArray(attrs)) {
         attrs.forEach(attr => this.element.setAttribute(attr.name, attr.val));
     } else {
-        Object.entries(attrs).forEach(([name,val]) => this.element.setAttribute(name, val));
+        Object.entries(attrs).forEach(([name, val]) => this.element.setAttribute(name, val));
     }
     return this;
 };
 
-o.prototype.class = function(classNames) {
+o.prototype.class = function (classNames) {
     if (Array.isArray(classNames)) {
         classNames.forEach(className => this.element.classList.add(className));
-    }else if (typeof classNames === 'string') {
+    } else if (typeof classNames === 'string') {
         this.element.className = classNames;
     }
     return this;
 };
 
-o.prototype.classList = function(classList) {
+o.prototype.classList = function (classList) {
     return this.class(classList)
 };
-o.prototype.className = function(className) {
+o.prototype.className = function (className) {
     return this.class(className)
 }
 
-o.prototype.id = function(id) {
+o.prototype.id = function (id) {
     this.element.setAttribute('id', id);
     return this;
 };
 
-o.prototype.add = function(children) {
-    if (this.element instanceof oFragment){
-        this.element.add(children);
-    }else {
-        children.forEach(child => {
-            try {
-                if(typeof child === 'boolean')
-                    return;
-                if (Array.isArray(child)){
-                    child.forEach(childEl => this.element.appendChild(childEl));
-                }else {
-                    this.element.appendChild(child);
-                }
-            }catch (err) {
-                console.warn(err);
-            }
-        });
+o.prototype.add = function (children) {
+    if (Array.isArray(children)) {
+        children.forEach(child => this.add(child));
+        return this;
+    }
+
+    if (children instanceof oFragment) {
+        children.children.forEach(child => this.add(child));
+        return this;
+    }
+
+    if (children instanceof HTMLElement) {
+        this.element.appendChild(children);
+    }
+
+    if (children instanceof o || children.__proto__.init) {
+        const oInstanceHTML = children.init();
+        if (oInstanceHTML instanceof HTMLElement) {
+            this.element.appendChild(oInstanceHTML);
+        }
     }
```

Comment:
Może dodać też return this w tych ifach. bo tak to bez potrzeby sprawdza kolejnego ifa

Reply:
Jak najbardziej

Reply:
W sumie to nie mam pojęcia, dlaczego tak to zostawiłem :(